src/iterators/linear.rs

```diff
@@ -6,20 +6,23 @@ use iterators::{HistogramIterator, PickyIterator};
 pub struct Iter<'a, T: 'a + Counter> {
     hist: &'a Histogram<T>,
 
-    valueUnitsPerBucket: i64,
-    currentStepHighestValueReportingLevel: i64,
-    currentStepLowestValueReportingLevel: i64,
+    // > 0
+    valueUnitsPerBucket: u64,
+    currentStepHighestValueReportingLevel: u64,
+    currentStepLowestValueReportingLevel: u64,
 }
 
 impl<'a, T: 'a + Counter> Iter<'a, T> {
     /// Construct a new linear iterator. See `Histogram::iter_linear` for details.
     pub fn new(hist: &'a Histogram<T>,
-               valueUnitsPerBucket: i64)
+               valueUnitsPerBucket: u64)
                -> HistogramIterator<'a, T, Iter<'a, T>> {
+        assert!(valueUnitsPerBucket > 0);
```

Comment:
Now that I think of it, no reason why I shouldn't provide an error message in the assert... Anyway, I think the performance impact will probably be minimal as it's only at the initialization of an iterator. The ones on `iterator/mod.rs:106` and the like are maybe of more concern, but even there I'm not too worried about it. I've found http://joeduffyblog.com/2016/02/07/the-error-model/#bugs-abandonment-assertions-and-contracts to be pretty persuasive about error handling in general, and it espouses a pro-assertion stance. 

I think for these sorts of assertions maybe `debug_assert!` is the configured the wrong direction by default. I want asserts on by default, even in release builds, and optionally removed for users who (1) know their code paths won't tickle the assertions and (2) need the last 2% of speed. We could have a `probably_assert!` (or whatever) that defaults to `assert!` unless a feature is set (`strip_asserts`?). I don't think we necessarily need this now, or maybe ever, but if benchmarks start showing assertions are painful, we have an easy out.